### ISSUE TITLE
Text in Safari dark mode not legible

### ISSUE THREAD

**Issue by mikehedman** [reactjs/reactjs.org#4201]
The explanatory text in dark mode in Safari is illegible:
![image](https://user-images.githubusercontent.com/1550290/147774829-68255413-5d11-4631-acc6-00efdd08fa72.png)
Switching Safari out of dark mode fixes it:
![image](https://user-images.githubusercontent.com/1550290/147774894-0df94fec-8bef-4487-9c0c-1ce731500fa6.png)

This was observed on an iPad Mini 6th Gen, iOS 15.0.1

**Comment by mikehedman** [comment]
Just a note - I've found that if I open the page by directly entering the URL in the address bar, it opens fine. But I can reliably reproduce the issue when navigating to the site through a Google search result.  I search for "react effect action", click on the "Hooks API Reference - React" link, and it opens bad.  But if I then refresh the page, it shows properly.

**Comment by mikehedman** [comment]
I've found that adding the following CSS rule properly renders the page in dark mode:
`:root {
    color-scheme: light dark;
}`
But it should be noted - it does render with light text on a dark background - which really is "dark mode". So that's good.  But if you then refresh the page, the text is legible, but it has dark text on a white background - so it's not properly rendering in dark mode.

**Comment by harish-sethuraman** [comment]
I tried to replicate the issue and failed to do so. We do not have a dark mode for the old reactjs.org site. Also, there is no code related to color schemes for the old site 🤔 . Do you have any extension that adds dark mode to sites by any chance?

**Comment by mikehedman** [comment]
@harish-sethuraman Thanks for taking a look.  No, I don't have any extensions installed.  The iPad is pretty new, so it's pretty much "stock".  I see that the site works the way it should all the time, except when it's loaded from search.  I spent quite a lot of time today trying to figure out why, and couldn't find anything.  When the page loads, the background is just black, with no CSS driving that.  WebKit isn't supposed to just force a site into dark mode, and with the site not specifying dark mode handling, what I'm seeing shouldn't be happening.  If I were to guess, it feels like maybe it's a Safari bug, somehow carrying forward (or, maybe more accurately not clearing out) the background color from the previous site (Google's search page, in this case).

**Comment by mikehedman** [comment]
@harish-sethuraman I think I've confirmed it's a Safari problem, and not a problem with the React doc site. I did a search for "mikehedman.com GitHub" (my own placeholder site), and when I selected it from the search results it also came up with a black background - obscuring the black text on the page. Refreshing the page resets the background to white (with black text).
I tried navigating to my site from a search via DuckDuckGo.com (their search page comes up in dark mode, whereas bing and yahoo don't), but it doesn't appear to have the same problem. Still feels like a Safari issue though, since the referring site should have no bearing on the display of a site.
I'm at iOS 15.0.1, so I'll try the 15.2 update to see if they happen to have fixed it.
